### PR TITLE
Update file format docs for Iceberg

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Removing some old stuff that is from Hive connector, and wrong in Iceberg.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/connector/iceberg.md [hunk 1]
```diff
@@ -681,13 +681,8 @@ TABLE </sql/show-create-table>`.
 
 Table properties supply or set metadata for the underlying tables. This is key
 for {doc}`/sql/create-table-as` statements. Table properties are passed to the
-connector using a {doc}`WITH </sql/create-table-as>` clause:
+connector using a {doc}`WITH </sql/create-table-as>` clause.
 
-```
-CREATE TABLE tablename
-WITH (format='CSV',
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:688]: This is definitely wrong for Iceberg...

File: docs/src/main/sphinx/connector/iceberg.md [hunk 2]
```diff
@@ -696,9 +691,10 @@ WITH (format='CSV',
 
   * - Property name
     - Description
-  * - ``format``
+  * - ``file_format``
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:694]: I am guessing..

File: docs/src/main/sphinx/connector/iceberg.md [hunk 3]
```diff
@@ -15,8 +15,8 @@ swap. The table metadata file tracks the table schema, partitioning
 configuration, custom properties, and snapshots of the table contents.
 
 Iceberg data files are stored in either Parquet, ORC, or Avro format, as
-determined by the `format` property in the table definition.  The default
-`format` value is `PARQUET`.
+determined by the `iceberg.file-format` property in the catalog properties file.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:18]: Maybe we need to also say that the table property is format and the column for the metadata is file_format .. its a mess..

File: docs/src/main/sphinx/connector/iceberg.md [hunk 4]
```diff
@@ -698,7 +693,8 @@ WITH (format='CSV',
     - Description
   * - ``format``
     - Optionally specifies the format of table data files; either ``PARQUET``,
-      ``ORC`, or ``AVRO``.  Defaults to ``ORC``.
+      ``ORC`, or ``AVRO``. Defaults to ``PARQUET`` from the default value
+      of ``iceberg.file-format`` or uses its setting from the catalog properties file.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:697]: Do we need to tie it to the file_format metadata?

File: docs/src/main/sphinx/connector/iceberg.md (hunk 1 above)
Comment (findinpath) [reply docs/src/main/sphinx/connector/iceberg.md:688]: That's good removal.
We have anyway an example of concrete table properties after the _Iceberg table properties_ table.

File: docs/src/main/sphinx/connector/iceberg.md [hunk 5]
```diff
@@ -718,7 +714,7 @@ WITH (format='CSV',
       Defaults to ``0.05``.
 ```
 
-The table definition below specifies format Parquet, partitioning by columns
+The table definition below specifies to use Parquet files, partitioning by columns
```
Comment (findinpath) [inline docs/src/main/sphinx/connector/iceberg.md:717]:
```suggestion
The table definition below specifies to use Parquet data files, partitioning by columns
```

File: docs/src/main/sphinx/connector/iceberg.md [hunk 6]
```diff
@@ -733,7 +729,7 @@ WITH (
     location = '/var/example_tables/test_table')
 ```
 
-The table definition below specifies format ORC, bloom filter index by columns
+The table definition below specifies to use ORC files, bloom filter index by columns
```
Comment (findinpath) [inline docs/src/main/sphinx/connector/iceberg.md:732]:
```suggestion
The table definition below specifies to use ORC data files, bloom filter index by columns
```

File: docs/src/main/sphinx/connector/iceberg.md (hunk 4 above)
Comment (findinpath) [reply docs/src/main/sphinx/connector/iceberg.md:697]: > Do we need to tie it to the file_format metadata?

I'm assuming here that you are referring to `$files` metadata table's `file_format` property.
In this case, my answer would be _no_.

Comment (findinpath) [inline docs/src/main/sphinx/connector/iceberg.md:697]: > or uses its setting from the catalog properties file.

WDYM?

File: docs/src/main/sphinx/connector/iceberg.md [hunk 7]
```diff
@@ -681,13 +681,8 @@ TABLE </sql/show-create-table>`.
 
 Table properties supply or set metadata for the underlying tables. This is key
 for {doc}`/sql/create-table-as` statements. Table properties are passed to the
-connector using a {doc}`WITH </sql/create-table-as>` clause:
```
Comment (findinpath) [inline docs/src/main/sphinx/connector/iceberg.md:684]: Unrelated: It feels a bit weird to have the _Table properties_ section after _ALTER TABLE SET PROPERTIES_ section which references some of the properties even though they haven been introduced.

File: docs/src/main/sphinx/connector/iceberg.md (hunk 4 above)
Comment (findinpath) [inline docs/src/main/sphinx/connector/iceberg.md:697]:
```suggestion
``ORC`, or ``AVRO``. Defaults to the value of ``iceberg.file-format`` 
configuration property which is by default ``PARQUET``.
```

File: docs/src/main/sphinx/connector/iceberg.md (hunk 3 above)
Comment (findinpath) [reply docs/src/main/sphinx/connector/iceberg.md:18]: I think we should rather stick to the old version of this sentence.
We can have `icebergb.file-format`   set to PARQUET, but create explicitly a table with `format` AVRO.

File: docs/src/main/sphinx/connector/iceberg.md (hunk 2 above)
Comment (huberty89) [reply docs/src/main/sphinx/connector/iceberg.md]: `format` was correct.

All Iceberg table properties are defined here: https://github.com/trinodb/trino/blob/b7e5ee55760e9c2b04214359bb21fd11f2c0f4c1/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableProperties.java#L41

File: docs/src/main/sphinx/connector/iceberg.md (hunk 3 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.md:18]: The old is not really correct either, but I am moving these details in later sections.

File: docs/src/main/sphinx/connector/iceberg.md (hunk 7 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.md:684]: Agreed.. but not going to fix that problem here..

File: docs/src/main/sphinx/connector/iceberg.md (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.md:697]: Updated

File: docs/src/main/sphinx/connector/iceberg.md (hunk 5 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.md:717]: We just say Parquet files and ORC files everywhere else .. should I update across the Iceberg connector docs?

**Comment by mosabua** [comment]
Better now @findinpath ?

File: docs/src/main/sphinx/connector/iceberg.md (hunk 5 above)
Comment (findinpath) [reply docs/src/main/sphinx/connector/iceberg.md:716]: > should I update across the Iceberg connector docs?

Probably not.
In the context above it should be obvious that we're speaking about data files.
Actually the end user shouldn't be concerned about the fact that there are as well metadata files.

**Review by findinpath — APPROVED** [review]

**Pull request merged**